src/main/java/com/uber/cadence/workflow/Workflow.java

```diff
@@ -589,7 +605,52 @@ public static CancellationScope newCancellationScope(Runnable runnable) {
   }
 
   /**
-   * Creates a CancellationScope that is not linked to a parent scope.
+   * Wraps a procedure in a CancellationScope. The procedure receives the wrapping CancellationScope
+   * as a parameter. Useful when cancellation is requested from within the wrapped code. The
+   * following example cancels the sibling activity on any failure.
+   *
+   * <pre><code>
+   *               Workflow.newCancellationScope(
+   *                   (scope) -> {
+   *                     Promise<Void> p1 = Async.proc(activities::a1).exceptionally(ex->
+   *                        {
+   *                           scope.cancel("a1 failed");
+   *                           return null;
+   *                        });
+   *
+   *                     Promise<Void> p2 = Async.proc(activities::a2).exceptionally(ex->
+   *                        {
+   *                           scope.cancel("a2 failed");
+   *                           return null;
+   *                        });
+   *                     Promise.allOf(p1, p2).get();
+   *                   })
+   *               .run();
+   * </code></pre>
+   *
+   * @param proc code to wrap in the cancellation scope
+   * @return wrapped proc
+   */
+  public static CancellationScope newCancellationScope(Functions.Proc1<CancellationScope> proc) {
```

Comment:
Do you think another API to create newDetachedCancellationScope makes sense which takes parent?

Reply:
It might theoretically. I don't envision any specific use case when it might be useful. Let's postpone its creation until a real use case comes up.